### PR TITLE
PWGCF: Lambda R2 Correlation

### PR DESCRIPTION
**Pull request by yashpatley**

### CODE REVIEW

**Review by BongHwi — DISMISSED** [review]

**Review by BongHwi — APPROVED** [review]

File: PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+
+    float nrapbins = static_cast<float>(cfg_nRapBins);
+    float kminrap = static_cast<float>(cfg_Rap_Min);
+    float kmaxrap = static_cast<float>(cfg_Rap_Max);
+    float nphibins = static_cast<float>(cfg_nPhiBins);
+    float kminphi = static_cast<float>(cfg_Phi_Min);
+    float kmaxphi = static_cast<float>(cfg_Phi_Max);
+
+    float rapbinwidth = (kmaxrap - kminrap) / nrapbins;
+    float phibinwidth = (kmaxphi - kminphi) / nphibins;
```
Comment (victor-gonzalez) [inline PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx:499]: This is a time consuming task for doing it for each pair specially because it is not needed
I would suggest doing it once at the `init` time and defining/storing the needed results at the task level

File: PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx [hunk 2]
```diff
@@ -0,0 +1,687 @@
...
+
+    float rap1 = p1.y();
+    float phi1 = p1.phi();
+
+    float rap2 = p2.y();
+    float phi2 = p2.phi();
```
Comment (victor-gonzalez) [inline PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx:505]: I would let the compiler to efficiently decide if temporal storage is needed

File: PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx [hunk 3]
```diff
@@ -0,0 +1,687 @@
...
+        // antilambda - antilambda
+        for (auto const& antilambda_2 : anti_lambda_tracks) {
+          if ((antilambda_2.index() != antilambda_1.index()) && (antilambda_1.postrackid() != antilambda_2.postrackid()) && (antilambda_1.negtrackid() != antilambda_2.negtrackid())) {
+            if (antilambda_2.mass() > lambda_left[0] && antilambda_2.mass() < lambda_left[1]) {
+              partPair<2, 2>(antilambda_1, antilambda_2);
+            }
+          }
+        }
+      }
+    }
```
Comment (victor-gonzalez) [inline PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx:666]: In here two of my previous comments apply

Using 0, 1, 2, ... specially when the same number has different meaning is confusing, error prone, difficult to maintain, and difficult to evolve
I would suggest to define an `enum` for each context, for instance
```
enum MassWindowType {
  kCentralWindow = 0,
  kLeftWindow,
  kRightWindow,
  kNoOfMassWindows
}
```
the latest one in case you consider the possibility to iterate across mass windows (and it is always a good practice), for the mass window context and the likes for single particles and pair combinations.

Then using those `enum`s as the corresponding parameter types of the templated functions will immediately clarify their use and avoid ambiguities and not easy to catch mistakes

Basically the same code is replicated several times and as said replicating code is error prone and difficult to maintain/evolve. I would suggest trying to parameterize what is needed to be replicated in order to not replicate it

**Comment by victor-gonzalez** [comment]
@yashpatley As this PR got incidentally approved and merged, please consider my comments in here for the next iteration
@BongHwi Thanks for ordering the merge but please, next time contact us

**Comment by BongHwi** [comment]
Hello @yashpatley @victor-gonzalez ,

Sorry for inconvenience! I overlooked it..

**Comment by yashpatley** [comment]
Hello @victor-gonzalez I will edit the code as per your suggestions and make a new PR. Thank you.

**Comment by victor-gonzalez** [comment]
@yashpatley Of course, in between, you will be able to run it on hyperloop to have a first view of your results with larger statistics
